Velaptor/Graphics/RenderMediator.cs

```diff
@@ -4,41 +4,224 @@
+        this.lineRenderBatchReactable = reactableFactory.CreateRenderLineReactable();
+
+        this.textureItemComparer = textureItemComparer;
+        this.fontItemComparer = fontItemComparer;
+        this.rectItemComparer = rectItemComparer;
+        this.lineItemComparer = lineItemComparer;
+
+        // Set all of the layers to a default value of the max value of int
+        for (var i = 0; i < this.allLayers.Length; i++)
+        {
+            this.allLayers.Span[i] = int.MaxValue;
+        }
     }
 
     /// <summary>
     /// Coordinates the rendering between each of the renderers.
     /// </summary>
     private void CoordinateRenders()
     {
-        this.pushReactable.Push(NotificationIds.RenderTexturesId);
-        this.pushReactable.Push(NotificationIds.RenderRectsId);
-        this.pushReactable.Push(NotificationIds.RenderFontsId);
-        this.pushReactable.Push(NotificationIds.RenderLinesId);
+        var textureItems = this.texturePullReactable.Pull(PullResponses.GetTextureItemsId);
+        var fontItems = this.fontPullReactable.Pull(PullResponses.GetFontItemsId);
+        var rectItems = this.rectPullReactable.Pull(PullResponses.GetRectItemsId);
+        var lineItems = this.linePullReactable.Pull(PullResponses.GetLineItemsId);
+
+        textureItems.Span.Sort(this.textureItemComparer);
+        fontItems.Span.Sort(this.fontItemComparer);
+        rectItems.Span.Sort(this.rectItemComparer);
+        lineItems.Span.Sort(this.lineItemComparer);
+
+        var layerIndex = 0;
+
+        for (var i = 0; i < textureItems.Length; i++)
+        {
+            var textureLayer = textureItems.Span[i].Layer;
+            if (this.allLayers.Span.Contains(textureLayer))
+            {
+                continue;
+            }
+
+            this.allLayers.Span[layerIndex] = textureLayer;
+            layerIndex++;
+        }
+
+        for (var i = 0; i < fontItems.Length; i++)
+        {
+            var fontLayer = fontItems.Span[i].Layer;
+            if (this.allLayers.Span.Contains(fontLayer))
+            {
+                continue;
+            }
+
+            this.allLayers.Span[layerIndex] = fontLayer;
+            layerIndex++;
+        }
+
+        for (var i = 0; i < rectItems.Length; i++)
+        {
+            var rectLayer = rectItems.Span[i].Layer;
+            if (this.allLayers.Span.Contains(rectLayer))
+            {
+                continue;
+            }
+
+            this.allLayers.Span[layerIndex] = rectLayer;
+            layerIndex++;
+        }
+
+        for (var i = 0; i < lineItems.Length; i++)
+        {
+            var lineLayer = lineItems.Span[i].Layer;
+            if (this.allLayers.Span.Contains(lineLayer))
+            {
+                continue;
+            }
+
+            this.allLayers.Span[layerIndex] = lineLayer;
+            layerIndex++;
+        }
+
+        this.allLayers.Span.Sort();
+
+        // Render all of the items in a coordinated fashion
```

Comment:
Should this be Render or Renders?
```suggestion
        // Renders all of the items in a coordinated fashion
```